metrics/web/test/common/presentation/toggle/widgets/toggle_test.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_switch/flutter_switch.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metrics_theme_data.dart';
+import 'package:metrics/common/presentation/toggle/theme/toggle_theme_data.dart';
+import 'package:metrics/common/presentation/toggle/widgets/toggle.dart';
+
+import '../../../../test_utils/metrics_themed_testbed.dart';
+
+void main() {
+  group("Toggle", () {
+    const inactiveColor = Colors.red;
+    const activeColor = Colors.blue;
+    const activeHoverColor = Colors.green;
+    const inactiveHoverColor = Colors.yellow;
+    const metricsTheme = MetricsThemeData(
+      toggleTheme: ToggleThemeData(
+        inactiveColor: inactiveColor,
+        activeColor: activeColor,
+        activeHoverColor: activeHoverColor,
+        inactiveHoverColor: inactiveHoverColor,
+      ),
+    );
+    final flutterSwitchFinder = find.byType(FlutterSwitch);
+    final mouseRegionFinder = find.ancestor(
+      of: flutterSwitchFinder,
+      matching: find.byType(MouseRegion),
+    );
+
+    testWidgets(
+      "throws an AssertionError if a value is null",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _MetricsSwitchTestbed(
+            value: null,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "applies the given value to the toggle widget",
+      (WidgetTester tester) async {
+        const value = true;
+
+        await tester.pumpWidget(const _MetricsSwitchTestbed(
+          value: value,
+        ));
+
+        final switchWidget = tester.widget<FlutterSwitch>(flutterSwitchFinder);
+
+        expect(switchWidget.value, equals(value));
+      },
+    );
+
+    testWidgets(
+      "applies the given on toggle callback to the toggle widget",
+      (WidgetTester tester) async {
+        // ignore: avoid_positional_boolean_parameters
+        void testCallback(bool value) {}
+
+        await tester.pumpWidget(_MetricsSwitchTestbed(
+          onToggle: testCallback,
+        ));
+
+        final switchWidget = tester.widget<FlutterSwitch>(flutterSwitchFinder);
+
+        expect(switchWidget.onToggle, equals(testCallback));
+      },
+    );
+
+    testWidgets(
+      "applies the inactive color when the toggle is off",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const _MetricsSwitchTestbed(
+          metricsThemeData: metricsTheme,
+          value: true,
```

Comment:
It seems like we do not need any specific value here to test that widget applies the parameter to another widget. The same for all tests below.
```suggestion
```